### PR TITLE
Add LM trainer

### PR DESCRIPTION
**Pull request by christofid**

### CODE REVIEW

**Review by drugilsberg — COMMENTED** [review]
Had only a quick look, will check better later, but I have some comments:
- what about calling the module `gt4sd.trainer.lm` and splitting the `lm` folder in `datasets` and `models`?
- update the year in the copyright notice
- I think we can safely install everything without relying on conda and relaxing a lot the boundaries on python and pip. Same goes for the requirements.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+    branches: [ "main", "test"]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (drugilsberg) [inline .github/workflows/tests.yaml:15]: do you think we can add "3.11"?

**Review by drugilsberg — COMMENTED** [review]
Great job, we should:
- add the license file (you can get it from the gt4sd-core updating the year)
- enable branch protection for `main`


Last thing, should we consider renaming `lm` to `transformers`? Most likely we can reuse everything for modalities beyond text.

File: README.md [hunk 2]
```diff
@@ -1 +1,72 @@
-# gt4sd-lm-trainer
\ No newline at end of file
+# GT4SD's Language Modeling trainer submodule
```
Comment (drugilsberg) [inline README.md:1]:
```suggestion
# GT4SD's trainer submodule for HF transformers and PyTorch Lightning
```

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+description = Transformers trainer submodule of GT4SD.
+author= GT4SD team
+long_description_content_type=text/markdown
+long_description = file: README.md
+python_requires = >= 3.7.*
+classifiers =
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Comment (drugilsberg) [inline setup.cfg:13]:
```suggestion
    Programming Language :: Python :: 3.8
    Programming Language :: Python :: 3.9
    Programming Language :: Python :: 3.10
    Programming Language :: Python :: 3.11
```

File: lm_example.jsonl [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{"source": "The present invention provides compounds suitable for use in the treatment of conditions where it is beneficial to halt bone loss and kill cancer cells, particularly in metastases to and primary tumours in the bone and surrounding tissues.", "target": "Consequently the present invention provides compounds comprising a bisphosphonate moiety linked to a phytochemical, pharmaceutical compositions thereof and methods of treatment of bone diseases and/or proliferative disorders.", "claim": "A compound of formula Q-T-L wherein Q is a bisphosphonate moiety, T is linker and L is an anti-osteolytic or osteoinductive phytochemical."}
```
Comment (drugilsberg) [inline lm_example.jsonl:1]: I would remove this file, keeping only the one inside `resources`

**Review by drugilsberg — CHANGES_REQUESTED** [review]
Great job, only minor comments. I would add the LICENSE file that is still missing.

File: lm_example.jsonl (hunk 4 above)
Comment (christofid) [reply lm_example.jsonl]: Done

File: README.md (hunk 2 above)
Comment (christofid) [reply README.md]: Done

**Comment by christofid** [comment]
@drugilsberg thanks for the review. I have addressed the majority of the requested changes. Some comments: the LICENCE file is already there in the main from the initial commit and there is no need to be added again. Python 3.11 is not supported at this point, but this can be addressed with a newer PR.

**Comment by drugilsberg** [comment]
> @drugilsberg thanks for the review. I have addressed the majority of the requested changes. Some comments: the LICENCE file is already there in the main from the initial commit and there is no need to be added again. Python 3.11 is not supported at this point, but this can be addressed with a newer PR.

great, sorry for the overlook but from the changes it wasn't apparent.

**Review by drugilsberg — APPROVED** [review]
Good to go

**Pull request merged**